common/lib/dependabot/pull_request_creator/message_builder/metadata_presenter.rb

```diff
@@ -0,0 +1,283 @@
+# frozen_string_literal: true
+
+require "dependabot/pull_request_creator/message_builder"
+
+module Dependabot
+  class PullRequestCreator
+    class MessageBuilder
+      class MetadataPresenter
+        extend Forwardable
+
+        attr_reader :dependency, :source, :metadata_finder,
+                    :vulnerabilities_fixed, :github_redirection_service
+
+        def_delegators :metadata_finder,
+                       :changelog_url,
+                       :changelog_text,
+                       :commits_url,
+                       :commits,
+                       :maintainer_changes,
+                       :releases_url,
+                       :releases_text,
+                       :source_url,
+                       :upgrade_guide_url,
+                       :upgrade_guide_text
+
+        def initialize(dependency:, source:, metadata_finder:,
+                       vulnerabilities_fixed:, github_redirection_service:)
+          @dependency = dependency
+          @source = source
+          @metadata_finder = metadata_finder
+          @vulnerabilities_fixed = vulnerabilities_fixed
+          @github_redirection_service = github_redirection_service
+        end
+
+        def to_s
+          msg = ""
+          msg += vulnerabilities_cascade
+          msg += release_cascade
+          msg += changelog_cascade
+          msg += upgrade_guide_cascade
+          msg += commits_cascade
+          msg += maintainer_changes_cascade
+          msg += break_tag unless msg == ""
+          "\n" + sanitize_links_and_mentions(msg, unsafe: true)
+        end
+
+        private
+
+        def vulnerabilities_cascade
+          return "" unless vulnerabilities_fixed&.any?
```

Comment:
Noice: reducing `vulnerabilities_fixed`+`metadata_finder` scope to just the target dependency!

Nit: the same varname representing a hash in `MessageBuilder` and an array here confused me a little. Probably meh for anyone who's _not_ reading both implementations side by side :trollface: